src/dash-table/dash/DataTable.js

```diff
@@ -166,6 +166,18 @@ export const propTypes = {
          */
         editable: PropTypes.bool,
 
+        /**
+         * If True, then the column is hideable through the UI.
+         * If there are multiple column headers, only the last row of
+         * the column headers will dispay the `hide` action button.
+         *
+         * Hidden columns are listed by id in the `hidden_columns` prop.
+         *
+         * If `hidden_columns` is not empty, a toggle button is displayed
+         * above the table that allows toggling columns visibility.
+         */
+        hideable: PropTypes.bool,
```

Comment:
Was briefly considering whether we want a global `hideable` or `columns_hideable` prop to set the default for all columns. I think perhaps not, as hiding *all* columns would be... bad. So we should encourage folks to think about which columns to allow hiding by not allowing the simple option of "hide them all."

Alternatively, if we have logic to prevent hiding the only remaining column (do we? 🙏) maybe a "hide them all" option wouldn't be so bad. What do you think?